### ISSUE TITLE
Suggestion: Add a cop to detect `public` methods that should be `private`

### ISSUE THREAD

**Issue by morissetcl** [rubocop/rubocop#10487]
Hello there,

First of all thanks for your work and sorry if the subject has already been posted but at first sight I did not find any topic about it.

On many projects I worked I noticed that so many times there is `public` methods that should be `private`.
And we end up with a bunch a of test on private methods.

Do you think it could be relevant to have a cop to detect this case?

```ruby
# bad
class MyAwesomeClass
  def initialize(args)
    ...
  end

  def call # method called outside
     my_method
  end

  def my_method # method called only inside the class
    anything
  end
end
```

```ruby
# good
class MyAwesomeClass
  def initialize(args)
    ...
  end

  def call # method called outside
     my_method
  end

  private

  def my_method # method called only inside the class
    anything
  end
end
```

I am looking forward to have your inputs on it.
Thanks.

**Comment by koic** [comment]
Unfortunately, static analysis cannot be able to determine if `my_method` is `public` (in other words, *Published Interface*) or `private`.
